ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryRoutingFunctionalTest.java

```diff
@@ -0,0 +1,398 @@
+        USER_TOPIC,
+        USER_PROVIDER,
+        VALUE_FORMAT,
+        timestampSupplier::getAndIncrement
+    );
+  }
+
+  @Before
+  public void setUp() {
+    //Create stream
+    makeAdminRequest(
+        REST_APP_0,
+        "CREATE STREAM " + USERS_STREAM
+            + " (" + USER_PROVIDER.ksqlSchemaString() + ")"
+            + " WITH ("
+            + "   kafka_topic='" + USER_TOPIC + "', "
+            + "   key='" + USER_PROVIDER.key() + "', "
+            + "   value_format='" + VALUE_FORMAT + "'"
+            + ");"
+    );
+    //Create table
+    output = KsqlIdentifierTestUtil.uniqueIdentifierName();
+    sql = "SELECT * FROM " + output + " WHERE ROWKEY = '" + KEY + "';";
+    List<KsqlEntity> res = makeAdminRequestWithResponse(
+        REST_APP_0,
+        "CREATE TABLE " + output + " AS"
+            + " SELECT COUNT(1) AS COUNT FROM " + USERS_STREAM
+            + " GROUP BY " + USER_PROVIDER.key() + ";"
+    );
+    QUERY_ID = extractQueryId(res.get(0).toString());
+    QUERY_ID = QUERY_ID.substring(0, QUERY_ID.length()-1);
+    waitForTableRows();
+    waitForStreamsMetadataToInitialize(
+        REST_APP_0, ImmutableList.of(host0, host1, host2), QUERY_ID);
+  }
+
+  @After
+  public void cleanUp() {
+    REST_APP_0.closePersistentQueries();
+    REST_APP_0.dropSourcesExcept();
+  }
+
+  @AfterClass
+  public static void classTearDown() {
+    TMP.delete();
+  }
+
+  @Test
+  public void shouldQueryActiveWhenActiveAliveQueryIssuedToStandby() {
+    // Given:
+    ClusterFormation clusterFormation = findClusterFormation(REST_APP_0, REST_APP_1, REST_APP_2);
+    waitForClusterToBeDiscovered(clusterFormation.standBy.right, 3);
+    sendHeartbeartsForWindowLength(
+        clusterFormation.standBy.right, clusterFormation.active.left, 2000);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.standBy.right, clusterFormation.active.left, HighAvailabilityTestUtil::remoteServerIsUp);
+
+    // When:
+    final List<StreamedRow> rows_0 = makePullQueryRequest(clusterFormation.standBy.right, sql);
+
+    // Then:
+    assertThat(rows_0, hasSize(HEADER + 1));
+    assertThat(rows_0.get(1).getRow(), is(not(Optional.empty())));
+    assertThat(rows_0.get(1).getRow().get().values(), is(ImmutableList.of(KEY, BASE_TIME, 1)));
+  }
+
+
+  @Test
+  public void shouldQueryActiveWhenActiveAliveStandbyDeadQueryIssuedToRouter() {
+    // Given:
+    ClusterFormation clusterFormation = findClusterFormation(REST_APP_0, REST_APP_1, REST_APP_2);
+    waitForClusterToBeDiscovered(clusterFormation.router.right, 3);
+    sendHeartbeartsForWindowLength(
+        clusterFormation.router.right, clusterFormation.active.left, 2000);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.right,
+        clusterFormation.active.left,
+        HighAvailabilityTestUtil::remoteServerIsUp);
+
+    // When:
+    final List<StreamedRow> rows_0 = makePullQueryRequest(clusterFormation.router.right, sql);
```

Comment:
how are we sure that the standby is dead here?

Reply:
I see you are waiting till route and active or up.. but standby could also be up by then. no?

Reply:
Standby is not heartbeating, it cannot be up